Library/Homebrew/formula_support.rb

```diff
@@ -28,13 +28,14 @@ def valid?
 
   def to_s
     return @explanation unless @explanation.empty?
+    os_name = OS::Mac.os_name(ENV["HOMEBREW_OSX_VERSION"])
```

Comment:
Sorry, this needs to use `ENV["HOMEBREW_OS_VERSION"]` (not `OSX_VERSION`) and be in the OS class so it works on non-OS X platforms. Alternatively, all the OS X references in code need to be moved into `extend/os`.